sklearn/utils/estimator_checks.py

```diff
@@ -1650,47 +1653,21 @@ def check_estimators_fit_returns_self(name, estimator_orig,
 def check_estimators_unfitted(name, estimator_orig):
     """Check that predict raises an exception in an unfitted estimator.
 
-    Unfitted estimators should raise either AttributeError or ValueError.
-    The specific exception type NotFittedError inherits from both and can
-    therefore be adequately raised for that purpose.
+    Unfitted estimators should raise a NotFittedError.
     """
 
     # Common test for Regressors, Classifiers and Outlier detection estimators
     X, y = _boston_subset()
 
     estimator = clone(estimator_orig)
 
-    msg = "fit"
-    if hasattr(estimator, 'predict'):
-        can_predict = False
-        try:
-            # some models can predict without fitting
-            # like GaussianProcess regressors
-            # in this case, we skip this test
-            pred = estimator.predict(X)
-            assert pred.shape[0] == X.shape[0]
-            can_predict = True
-        except ValueError:
-            pass
-        if can_predict:
-            raise SkipTest(
-                "{} can predict without fitting, skipping "
-                "check_estimator_unfitted.".format(name))
-
-        assert_raise_message((AttributeError, ValueError), msg,
-                             estimator.predict, X)
-
-    if hasattr(estimator, 'decision_function'):
-        assert_raise_message((AttributeError, ValueError), msg,
-                             estimator.decision_function, X)
-
-    if hasattr(estimator, 'predict_proba'):
-        assert_raise_message((AttributeError, ValueError), msg,
-                             estimator.predict_proba, X)
-
-    if hasattr(estimator, 'predict_log_proba'):
-        assert_raise_message((AttributeError, ValueError), msg,
-                             estimator.predict_log_proba, X)
+    msg = "not fitted yet. Call 'fit' with appropriate arguments"
```

Comment:
What do you think of my critique of requiring this message? I'd be happy to require NotFittedError without constraining the message

Reply:
I am fine with it as well. Enforcing the message is maybe not necessary by the end.